inst/rmarkdown/templates/template_sfp/skeleton/skeleton.Rmd

```diff
@@ -0,0 +1,109 @@
+---
+title: {{{title}}}
+subtitle: {{{subtitle}}}
+author: {{{authors}}}
+date: {{{date}}}
+output: 
+  bookdown::html_document2:
+    toc: true
+    toc_float: true
+    number_sections: true
+---
+
+<!--
+Instructies bij opmaak van veldprotocol:
+
+Hieronder vind je de verschillende onderdelen van een veldprotocol en een beschrijving wat er bij elk onderdeel vermeld moet worden. 
+Niet alle onderdelen zijn altijd van toepassing. Laat in die gevallen de titelhoofdingen staan en vermeld expliciet “niet van toepassing”.
+Probeer het veldprotocol generiek te maken in de mate van mogelijk en probeer te vermijden dat het veldprotocol enkel specifiek voor een bepaald project bruikbaar is. 
+Je kan het protocol generiek houden door bepaalde variabelen, zoals bijvoorbeeld de grootte van een proefvlak of de diepte van een bodembemonstering, niet vast te leggen in het protocol. 
+In bijlage kan je dan een tabel toevoegen waarin je per project aangeeft welke waardes gekozen moeten worden voor die variabelen. 
+Indien een bepaald onderdeel niet van toepassing is, laat je de hoofding staan en vermeld je hieronder ‘Niet van toepassing’. Op die manier is de nummering van alle protocols dezelfde.
+-->
+
+```{r setup, include=FALSE}
+library(knitr)
+opts_chunk$set(
+  echo = FALSE, 
+  eval = TRUE,
+  dpi = 300,
+  fig.width = 150 / 25.4,
+  fig.height = 100 / 25.4,
+  out.width = '100%',
+  warning = FALSE,
+  error = TRUE,
+  message = FALSE
+)
+```
+
+
+```{r metadata}
+metadata <- data.frame(
+  Reviewers = "{{{reviewers}}}",
+  Documentbeheerder = "{{{file_manager}}}",
+  Procedure = "{{{procedure}}}",
+  Revisie = "{{{revision}}}"
+)
+
+kable(metadata)
+
+```
+
+```{r dependencies, child = "_01_dependencies.Rmd"}
```

Comment:
do you need this a child documents? Can't you use files like `01_dependencies.Rmd` which are included automatically by bookdown?

Reply:
:smile: yes I can, see what I wrote here: [PR message](https://github.com/inbo/protocolshelper/pull/1#issue-389923742).

But do you have compelling reasons to do so?

Reply:
Not having to write the child chunks seem a good reason to me.

Note that you could see if you can create a `protocol.Rproj` file from a template in the target folder which sets up a project for each protocol.

Reply:
This is the content of a `.Rproj` with a bookdown at the root.
```
Version: 1.0

RestoreWorkspace: No
SaveWorkspace: No
AlwaysSaveHistory: No

EnableCodeIndexing: Yes
UseSpacesForTab: Yes
NumSpacesForTab: 2
Encoding: UTF-8

RnwWeave: knitr
LaTeX: XeLaTeX

AutoAppendNewline: Yes
StripTrailingWhitespace: Yes

BuildType: Website

QuitChildProcessesOnExit: Yes
```

Reply:
A (better?) alternative to the multiple .Rproj files is to use `rprojroot` (or `here`) package consistently throughout. Although there is a little bit more code involved, I think this is a safer way of working (no risk of opening the wrong Rproj and wondering why that bookdown is not building). 

If you use something like 
````
my_sfp_root <- here::here("src", "thematic", "vegetation", "my-sfp")
my_table <- file.path(my_sfp_root, "blabla.csv")
kable(my_table)
````
inside Rmarkdown files, compilation of the bookdown will run without problem even if the location of the .Rproj is different from the location of the .Rmd files.

Reply:
My personal workflow is to have a dedicated `.Rproj` for each bookdown with the name of the matching with the folder name. So if I want to work on protocol A, I'd open `protocol_A.Rproj`. Switching to another protocol = switching to another RStudio project.
The root of the project = root of the bookdown. So no need for complicated `here()` statements. Everything should be relative to the root of the bookdown project.

Reply:
@ThierryO that's my approach as well, but it may not be compatible with any desired directory setup as it requires the Rmd files to be in the project's root. Still more importantly, it requires manual switching of RStudio project which is easily forgotten and a barrier to users. Actually I support a more general approach as the aim is to be user-friendly, even when that takes a little more coding (I should learn it too).

Reply:
But @hansvancalster I would use simpler coding such as `here::here("src/thematic/vegetation/my-sfp)"`. Forward slashes will always work AFAIK.

Reply:
I'm going to stick to the `rprojroot` approach.